src/Mod/Part/Gui/TaskCheckGeometry.cpp

```diff
@@ -433,11 +433,22 @@ void TaskCheckGeometryResults::goCheck()
     for(const auto &sel :  selection) {
         selectedCount++;
         TopoDS_Shape shape = Part::Feature::getShape(sel.pObject,sel.SubName,true);
-        if (shape.IsNull())
+        if (shape.IsNull()) {
+            Base::Console().Warning(tr("CheckGeometry: Skipping %1, has null or no shape.")
+                                    .arg(QString::fromUtf8(sel.pObject->getNameInDocument())).toStdString().c_str());
+            Base::Console().Message("\n");
             continue;
+        }
+        if (Part::TopoShape(shape).getBoundBox().CalcDiagonalLength() > 1e+100) {
```

Comment:
Would it be possible to use `shape.Infinite()` instead? like this:
```suggestion
        if (shape.Infinite()) {
```